### PR TITLE
Page Header: Update markup, support custom icons, and Breadcrumbs

### PR DESCRIPTION
**Pull request by kevinparkerson**
Fixes #1586 (expanded), #1972, #1716, and #1984 

Replaces #1687 for this effort.

* Markup is all updated, and the components have all been structured better. 
* Support for icons & breadcrumbs has been added (Setup example), 
*  Anew prop to be used with `object-home` called `nameSwitcherDropdown` has been added to better reflect the SLDS structure. 
* `onRenderActions` and `onRenderContent` now consistently refers to the same items across variants.
* Also added a SLDSPageHeaderControl component to handle the `.slds-page-header__control` wrapper as discussed.

---

### CONTRIBUTOR checklist (do not remove)
Please complete for every pull request

* [x] First-time contributors should sign the Contributor License Agreement. It's a fancy way of saying that you are giving away your contribution to this project. If you haven't before, wait a few minutes and a bot will comment on this pull request with instructions.
* [x] `npm run lint:fix` has been run and linting passes.
* [x] Mocha, Jest (Storyshots), and `components/component-docs.json` CI tests pass (`npm test`).
* [x] Tests have been added for new props to prevent regressions in the future. See [readme](https://github.com/salesforce/design-system-react/blob/master/tests/README.md).
* [x] Review the appropriate Storybook stories. Open [http://localhost:9001/](http://localhost:9001/).
* [x] Review tests are passing in the browser. Open [http://localhost:8001/](http://localhost:8001/).
* [x] Review markup conforms to [SLDS](https://www.lightningdesignsystem.com/) by looking at [DOM snapshot strings](https://facebook.github.io/jest/docs/en/snapshot-testing.html).

### REVIEWER checklist (do not remove)

* [ ] TravisCI tests pass. This includes linting, Mocha, Jest, Storyshots, and `components/component-docs.json` tests.
* [ ] Tests have been added for new props to prevent regressions in the future. See [readme](https://github.com/salesforce/design-system-react/blob/master/tests/README.md).
* [ ] Review the appropriate Storybook stories. Open [http://localhost:9001/](http://localhost:9001/).
* [ ] The Accessibility panel of each Storybook story has 0 violations (aXe). Open [http://localhost:9001/](http://localhost:9001/).
* [ ] Review tests are passing in the browser. Open [http://localhost:8001/](http://localhost:8001/).
* [ ] Review markup conforms to [SLDS](https://www.lightningdesignsystem.com/) by looking at [DOM snapshot strings](https://facebook.github.io/jest/docs/en/snapshot-testing.html).
###### Required only if there are markup / UX changes
* [ ] Add year-first date and commit SHA to `last-slds-markup-review` in `package.json` and push.
* [ ] Request a review of the deployed Heroku app by the Salesforce UX Accessibility Team.
* [ ] Add year-first review date, and commit SHA, `last-accessibility-review`, to `package.json` and push.
* [ ] While the contributor's branch is checked out, run `npm run local-update` within locally cloned [site repo](https://github.com/salesforce-ux/design-system-react-site) to confirm the site will function correctly at the next release.

### CODE REVIEW

**Comment by interactivellama** [comment]
Please assign the PR to me when it is ready for review @kevinparkerson

**Comment by interactivellama** [comment]
To much spacing on https://design-system-react-co-pr-2011.herokuapp.com/?selectedKind=SLDSPageHeader&selectedStory=Base%20with%20controls&full=0&addons=1&stories=1&panelRight=0&addonPanel=storybook%2Factions%2Factions-panel

Screen Shot 2019-06-07 at 2 32 34 PM

**Comment by interactivellama** [comment]
Why are these gray? https://design-system-react-co-pr-2011.herokuapp.com/?selectedKind=SLDSPageHeader&selectedStory=Object%20Home&full=0&addons=1&stories=1&panelRight=0&addonPanel=storybook%2Factions%2Factions-panel

I think they are the wrong variant.

Screen Shot 2019-06-07 at 2 34 43 PM

**Comment by interactivellama** [comment]
It looks like Breadcrumb no longer has bold text in it. https://lightningdesignsystem.com/components/breadcrumbs/#Base

That's probably the issue here:

https://design-system-react-co-pr-2011.herokuapp.com/?selectedKind=SLDSPageHeader&selectedStory=Related%20List&full=0&addons=1&stories=1&panelRight=0&addonPanel=storybook%2Factions%2Factions-panel
Screen Shot 2019-06-07 at 4 03 27 PM

https://lightningdesignsystem.com/components/page-headers/#Related-List
Screen Shot 2019-06-07 at 4 03 41 PM

**Comment by interactivellama** [comment]
https://lightningdesignsystem.com/components/page-headers/#Related-List
Screen Shot 2019-06-07 at 4 11 30 PM

meta info section font size looks too small:
Screen Shot 2019-06-07 at 4 11 26 PM
https://design-system-react-co-pr-2011.herokuapp.com/?selectedKind=SLDSPageHeader&selectedStory=Related%20List&full=0&addons=1&stories=1&panelRight=0&addonPanel=storybook%2Factions%2Factions-panel

File: components/breadcrumb/index.jsx [hunk 1]
```diff
@@ -70,6 +70,10 @@ Breadcrumb.propTypes = {
 	assistiveText: PropTypes.shape({
 		label: PropTypes.string,
 	}),
+	/**
+	 * Custom styles to be passed to the top-level `nav` tag
+	 */
+	style: PropTypes.object,
```
Comment (interactivellama) [inline components/breadcrumb/index.jsx]: This should be named something else. `styleNav` or `styleContainer`

`style` and `className` should only be used when applied to `.slds-breadcrumb`

**Comment by interactivellama** [comment]
@kwilloughby This has the Setup PageHeader that you requested in it and is viewable.

Screen Shot 2019-06-07 at 4 20 03 PM

File: components/page-header/check-props.js [hunk 2]
```diff
@@ -53,13 +59,58 @@ if (process.env.NODE_ENV !== 'production') {
...
 		deprecatedProperty(
 			COMPONENT,
 			props.navRight,
 			'navRight',
 			'onRenderControls',
 			createDocUrl('onRenderControls')
 		);
+
+		if (props.onRenderActions) {
+			renderFunctionReturnContentsLackDisplayName(
```
Comment (interactivellama) [inline components/page-header/check-props.js]: Great way to add user affordance and go the extra mile for our consumers @kevinparkerson !

**Comment by kevinparkerson** [comment]
@interactivellama updated everything you commented on thus far

File: components/breadcrumb/index.jsx (hunk 1 above)
Comment (kevinparkerson) [reply components/breadcrumb/index.jsx]: Done!

**Comment by interactivellama** [comment]
@kevinparkerson 

http://localhost:9001/?selectedKind=SLDSSplitView&selectedStory=Base%3A%20Open&full=0&addons=1&stories=1&panelRight=0&addonPanel=storybook%2Factions%2Factions-panel
Screen Shot 2019-06-26 at 4 48 32 PM

See
http://design-system-react-components.herokuapp.com/?selectedKind=SLDSSplitView&selectedStory=Base%3A%20Open&full=0&addons=1&stories=1&panelRight=0&addonPanel=storybook%2Factions%2Factions-panel for comparison.

This might mean that the old API `navRight` and `contentRight` aren't working.

**Comment by kevinparkerson** [comment]
@interactivellama this is ready for re-review now!

**Comment by kwilloughby-sf** [comment]
@kevinparkerson & @interactivellama, looking good. Don't forget about this issue.

https://github.com/salesforce/design-system-react/issues/2035

**Comment by kevinparkerson** [comment]
@kwilloughby-sf thanks, almost forgot about that one! I'll add it in a separate PR

**Pull request merged**